src/graphql/entries.ts

```diff
@@ -77,18 +78,76 @@ export async function updateEntry({
   return copyOfDataFromPreviewApp;
 }
 
-function updateRichTextField({
+interface RichTextLink {
+  block: (Entity & CollectionItem)[];
+  inline: (Entity & CollectionItem)[];
+}
+
+function isEntityLinkEmpty(obj: RichTextLink) {
+  return Object.values(obj).every((arr) => arr.length === 0);
+}
+
+async function updateRichTextField({
   dataFromPreviewApp,
   updateFromEntryEditor,
   name,
   locale,
+  entityReferenceMap,
 }: UpdateFieldProps) {
   if (name in dataFromPreviewApp) {
     if (!dataFromPreviewApp[name]) {
       dataFromPreviewApp[name] = {};
     }
+
+    // Update the rich text JSON data
     (dataFromPreviewApp[name] as { json: unknown }).json =
       updateFromEntryEditor?.fields?.[name]?.[locale] ?? null;
+
+    // Update the rich text embedded entries
+    // Initialize arrays to hold the resolved entries and assets, separated by block and inline
+    const entries: RichTextLink = { block: [], inline: [] };
+    const assets: RichTextLink = { block: [], inline: [] };
+
+    // Iterate over each node in the rich text content
+    for (const node of dataFromPreviewApp[name].json.content) {
+      // Check if the node is an embedded entity
+      if (node.nodeType.includes('embedded')) {
+        if (node.data && node.data.target && node.data.target.sys) {
+          const id = node.data.target?.sys.id || '';
+          const updatedReference = {
+            sys: { id: id, type: 'Link', linkType: node.data.target.sys.type },
+          };
+          // Use the updateReferenceEntryField function to resolve the entity reference
+          const ref = await updateReferenceEntryField(
+            null,
+            updatedReference,
+            entityReferenceMap,
+            locale
+          );
+          // Depending on the node type, assign the resolved reference to the appropriate array
+          switch (node.nodeType) {
```

Comment:
Does this mean we will need to implement also resource links here? 🤔

Reply:
I don't think we support resource links as embedded entries in rich text yet right?
But once we do I guess we should test it (I think resource links open in a new tab so not sure if we need to do anything)

Reply:
I'll create a follow-up ticket for this then, I think we should try to merge this one in sooner rather than later since it's a few users have brought it up already